File: web-dev/cs50/intro-comp-science/Problem-sets/stuff/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
int LEN = 4;
//void least_coins(int x);


int main(void)
{
    int coins[4] = {25, 10, 5, 1};
    float dollar;
    int count;
    int number = 0;
    // This prompts the user of what amount he/she is expecting
    do
    {
        dollar = get_float("Change owed: ");
    }
    // The condition that that the user must fulfil before the algorithm will execute. 
    while (dollar < 0);
    
    // Converting the Dollar value into coin value.
    int change = round(dollar * 100);

    // Creating our for-loop to calculate the amount to be given without doing it one-by-one
    for (int i = 0; i < LEN; i++)

    {
        if (coins[i] <= change)
        {
            // This will loop throught the set of coins as we reduce the amount to be give out 
            count = change / coins[i];
            // This saves the number of times we have used some value of coins and saves it 
            number = number + count;
            // This keeps track of the remaining change to be given to the customer
            change = change - count * coins[i];
        }
    }
    printf("%i\n", number);
}

File: web-dev/cs50/intro-comp-science/Problem-sets/pset6/cash.py
from cs50 import get_float #importing the get_float funtion from cs50 libraries


# Defining our main funtion for the program
def main():
    # Setting the condition that the user must meet if not it will keep asking until right input is given
    while True:
        ## Getting input from the user
        dollar = get_float("Change owed: ")
        # The input should be a non-negative integer
        if dollar >= 0:
            break
    # Changing the dollar input into cents
    change = int(dollar * 100)

    # Calling the function to calculate the amount
    calc(change)

# defining our function to calculate the least number of coins to be given
def calc(change):
    number = 0
    coins = [25, 10, 5, 1]
    for coin in coins:
        if coin <= change:
            count = change // coin
            number += count
            change = change - count * coin
    print(f"{number}\n", end=" ")

    # Calling back main
main()
